### PR TITLE
Fix 2431 - inconsistent naming of json methods and missing json varargs {} method

### PR DESCRIPTION
**Pull request by maxgrenderjones**
See #2431

### CODE REVIEW

File: lib/pure/json.nim [hunk 1]
```diff
@@ -723,22 +723,25 @@ proc `[]=`*(obj: JsonNode, key: string, val: JsonNode) =
...
-  ## Transverses the node and gets the given value. If any of the
-  ## names does not exist, returns nil
+proc `{}`*(node: JsonNode, keys: varargs[string]): JsonNode =
+  ## Traverses the node and gets the given value. If any of the
+  ## keys do not exist, returns nil
   result = node
-  if isNil(node): return nil
-  result = result[key]
+  for key in keys:
+    if isNil(result) or not result.hasKey(key):
```
Comment (Araq) [inline lib/pure/json.nim]: hasKey and [] are 2 lookups when you can do the same with 1 lookup.

**Comment by maxgrenderjones** [comment]
Thanks for the feedback @Araq, now only does one lookup. Couple of questions:
 - Assume that the purpose of {} is to do always-safe indexing, so it's right to return nil if someone tries to index into a non-object (rather than throw an error)
 - I opened the pull request against master - should I change it to be against devel?

**Comment by Araq** [comment]
* Dunno, what was the old behaviour?
* Oh yes of course! Against devel!

**Comment by maxgrenderjones** [comment]
Superceded by #2572

**Pull request closed without merging**